File: 4115-minimum-distance-between-three-equal-elements-i/4115-minimum-distance-between-three-equal-elements-i.java
class Solution {
    public int minimumDistance(int[] nums) {
        int n = nums.length;
        int mini = Integer.MAX_VALUE;
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                for (int k = j + 1; k < n; k++) {
                    if (nums[i] == nums[j] && nums[j] == nums[k]) {
                        mini = Math.min(mini, Math.abs(i - j) + Math.abs(j - k) + Math.abs(k - i));
                    }
                }
            }
        }
        return mini == Integer.MAX_VALUE ? -1 : mini;
    }
}

File: 4115-minimum-distance-between-three-equal-elements-i/4115-minimum-distance-between-three-equal-elements-i.py
class Solution:
    def minimumDistance(self, nums: List[int]) -> int:
        n = len(nums)
        mini = (10 ** 9)
        for i in range(n):
            for j in range(i + 1, n):
                for k in range(j + 1, n):
                    if nums[i] == nums[j] and nums[j] == nums[k] and nums[k] == nums[i]:
                        mini = min(mini, abs(i - j) + abs(j - k) + abs(k - i))

        if mini == (10 ** 9): return -1
        return mini
        